src/UnigramWordPredictor.java

```diff
@@ -52,6 +53,26 @@ public void train(Scanner scanner) {
     List<String> trainingWords = tokenizer.tokenize(scanner);
 
     // TODO: Convert the trainingWords into neighborMap here
+    //initialize map (we in this case cannot do this within the 
+    //constructor, so we can point neighborMap within method to
+    //a new HashMap<>()
+    this.neighborMap = new HashMap<>();
```

Comment:
This works well! You can make it a bit simpler by omitting the `this`.
```
neighborMap = new HashMap<>();
```
It does the same thing: for any variables that are not otherwise declared in the local scope, Java assumes they are instance variables and essentially puts an invisible `this.` in front of them.